### DISCUSSION TITLE
Test-drive workflow-kotlin in a fragment-based app?

### DISCUSSION THREAD

**Discussion by tudorluca** [square/workflow-kotlin#271] [Q&A] [answered]
I'd like to take workflow-kotlin for a test-drive in an existing app, heavily based on Fragments.

Would it be OK if I use workflow-kotlin to just manage the state of an `(Android)ViewModel`, without the UI stuff? basically a Workflow per Fragment, that handles all the plumbing for that screen.

**Comment by zach-klippenstein** [comment]
Yes! The UI stuff is completely optional, and it should be fairly straightforward to use the non-UI api in this way. See `renderWorkflowIn`: https://square.github.io/workflow/kotlin/api/workflow/com.squareup.workflow1/render-workflow-in/

**Comment by zach-klippenstein** [comment, marked as answer]
From @rjrjr:

> You might find the UI module’s new `renderWorkflowIn` overload to be just the thing here. It ties into the androidx `ViewModel` system for lifecycle and persistence, but doesn’t dictate anything else. You can use that and completely ignore `WorkflowLayout`, `ViewEnvironment`, etc.
>
> Basically imitate `HelloWorkflowFragment`, but return your own view instance instead of ours.
> https://github.com/square/workflow-kotlin/blob/v1.0.0-alpha.9/samples/hello-workfl[…]/squareup/sample/helloworkflowfragment/HelloWorkflowFragment.kt
